src/binary-search-tree.js

```diff
@@ -5,6 +5,49 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(value) {
+    const node = new BinarySearchTree(value);
+    function recurse(bst) {
+      if (bst.value > value && bst.left === null) {
+        bst.left = node;
+      } else if (bst.value > value) {
+        recurse(bst.left);
+      } else if (bst.value < value && bst.right === null) {
+        bst.right = node;
+      } else if (bst.value < value) {
+        recurse(bst.right);
+      }
+    }
+    recurse(this);
+  }
+
+  contains(value) {
+    let doesContain = false;
+    function recurse(bst) {
```

Comment:
Same deal here, your recursive function doesn't return. Think about the base case more carefully.